client/components/Assignments/AssignmentItem/fields/Headline.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+interface IProps {
+    assignment: any;
+    loadArchiveItem: (assignment) => any;
+}
+
+export const HeadlineComponent = ({assignment, loadArchiveItem}: IProps) => {
+    const itemIds = assignment.item_ids;
+
+    if (!itemIds) {
+        return null;
+    }
+
+    const itemId = assignment.item_ids[0];
+
+    if (!itemId) {
+        return null;
+    }
+
+    const [item, setItem] = React.useState(null);
+
+    if (!item) {
+        loadArchiveItem(assignment).then(setItem);
```

Comment:
Any ideas on how to handle changes to archive item's headline?

Reply:
I guess we should tackle this like the `name` field approach, data changes should trigger a re-render of the parent component, and the child shouldn't be aware of data changes. Do we use events like `content:update` or so in planning?

Reply:
Below is a suggested solution to this problem, including reducing performance impact of loading archive items.

Keep in mind that loading/syncing archive items should only happen if any field that requires archive items, such as `headline`, is configured in `appConfig.assignmentsList`.

**1. Bulk loading required archive items**
Instead of sending an API query for every Assignment that has content linked, it might be best to send a single query to fetch all associated content when receiving assignments.

Create a new dispatch action for assignments that bulk loads archive items (similar to the existing `loadArchiveItem`, https://github.com/superdesk/superdesk-planning/blob/master/client/actions/assignments/api.js#L485), and dispatch that action when receiving assignments at https://github.com/superdesk/superdesk-planning/blob/master/client/actions/assignments/api.js#L263.

**2. Keeping archive items up to date**
To listen for angular broadcast events in Planning you can add a new entry into the respective actions/notifications file.

i.e.
Create an `onContentUpdated` function that re-fetches the archive item and updates the store if the item exists in `selectors.assignments.getStoredArchiveItems(getState()`)

Add this function to the assignment notifications map in https://github.com/superdesk/superdesk-planning/blob/master/client/actions/assignments/notifications.js#L376:

``` js
'content:update': () => (self.onContentUpdated),
```
This should keep any currently loaded archive item up to date.

**3. Sending the archive item to the AssignmentItem component**
Add `selectors.assignments.getStoredArchiveItems(state)` to the `AssignmentGroupList` component props, which can then be passed down to the `AssignmentItem` and automatically updated if/when the store updates.